[Accessibility] docs (#17698)
backstage/backstage: docs/accessibility/index.md

---
id: index
title: Backstage Accessibility
description: Documentation on Backstage Accessibility
---

In an effort to bake accessibility practices further into our process of building features for Backstage, we’ve rolled out automated CI tests to the OSS project on some of our core features, Software Catalog, Software Templates, Search and TechDocs. As these are just a few plugins of many out there, we want to encourage you to consider the accessibility implications of your work on Backstage in order to build a great experience for everyone.

## How to contribute

> There are multiple ways to contribute to making Backstage accessible to everyone, below we list examples of some ways you can do that.

Reviewer: ```suggestion
There are multiple ways to contribute to making Backstage more accessible, you'll find below a list of examples to help you get started.
```

### Run Lighthouse in CI on your plugin

> If your plugin lives in the [Backstage main repository](https://github.com/backstage/backstage/) you can modify the [urls in the Lighthouse config](https://github.com/backstage/backstage/blob/master/lighthouserc.js#L19-L34) to run the Lighthouse checks on urls where your plugin exists as well. E.g.

Reviewer 2: To make sure we're referencing lines which aren't subject to drift up and down the file, suggest to point to a specific git  hash rather than a branch. There was recently a PR to refactor that across the docs so it'll help to keep consistent!

Author: good call out, as we just discussed that in the thread above and I wanted to at least keep the line specific links as permalinks! Will update!

Author: updated in https://github.com/backstage/backstage/pull/17698/commits/1d2932d97194db66a663e5a970c78fa8330718f5

```diff
 ci: {
    collect: {
      url: [
+        /** Your plugin paths */
+        'http://localhost:3000/your-plugin-path,
        /** Software Catalog */
        'http://localhost:3000/catalog',
        'http://localhost:3000/catalog-import',
        'http://localhost:3000/catalog/default/component/backstage',
        ...
      ],
      settings: {
       ...
      },
     ...
    },
    assert: {
      ...
    },
  },
```

> To make sure the [Accessibility Github workflow](https://github.com/backstage/backstage/blob/master/.github/workflows/verify_accessibility.yml) is running when changes are made to your plugin folders, also modify the [list of paths](https://github.com/backstage/backstage/blob/10759b6ad2561bd86183ad940256f9a309c7a6b0/.github/workflows/verify_accessibility.yml#L7-L16).

Reviewer: ```suggestion
To make sure the [Accessibility Github workflow](https://github.com/backstage/backstage/blob/master/.github/workflows/verify_accessibility.yml) is running when changes are made to your plugin folders, modify the [list of paths](https://github.com/backstage/backstage/blob/10759b6ad2561bd86183ad940256f9a309c7a6b0/.github/workflows/verify_accessibility.yml#L7-L16).
```

> To make sure the [Accessibility Github workflow](https://github.com/backstage/backstage/blob/10759b6ad2561bd86183ad940256f9a309c7a6b0/.github/workflows/verify_accessibility.yml) is running when changes are made to your plugin folders, also modify the [list of paths](https://github.com/backstage/backstage/blob/10759b6ad2561bd86183ad940256f9a309c7a6b0/.github/workflows/verify_accessibility.yml#L7-L16).

Reviewer 3: IDEA: Maybe change this link to master. https://github.com/backstage/backstage/blob/10759b6ad2561bd86183ad940256f9a309c7a6b0/.github/workflows/verify_accessibility.yml -> https://github.com/backstage/backstage/blob/master/.github/workflows/verify_accessibility.yml .
Also applicable to the other link specifying lines 7 to 16.

Author: Yeah its great points, I decided to used permalinks in case the line numbers changed (e.g. new path gets added) to avoid pointing to the wrong part of the config. But let me update the one that don't use line number 👍

Author: https://github.com/backstage/backstage/pull/17698/commits/ee740ac2eac79cc6f4ebabff0d9665b2562beffb

### Run the Lighthouse CLI locally when developing new features

If you want to use the Lighthouse CLI and run the checks based on the config you can use the following command:

```
yarn dlx @lhci/cli@0.11.x autorun
```

> Note: running this command will use the [Lighthouse config](https://github.com/backstage/backstage/blob/master/lighthouserc.js#L19-L34) so make sure to adjust it to your needs if needed.

### Use Lighthouse Github Action on your own repo

> If your Backstage plugin lives outside of the [Backstage main repository](https://github.com/backstage/backstage/), and you use Github Actions for continuous integration, we encourage you to add and modify the [Accessibility Github workflow](https://github.com/backstage/backstage/blob/master/.github/workflows/verify_accessibility.yml) to your needs.

Reviewer: ```suggestion
If your Backstage plugin lives outside of the [Backstage main repository](https://github.com/backstage/backstage/), and you use GitHub Actions for continuous integration, we encourage you to add and modify the [Accessibility GitHub workflow](https://github.com/backstage/backstage/blob/master/.github/workflows/verify_accessibility.yml) to your needs.
```

### Report identified issues

> It’s important to remember that the automated checks can just catch a very low number of accessibility issues, therefore we encourage you to do manual testing of your plugins using Assistive technology (e.g. Screen readers, alternative navigation, screen magnifiers, to mention a few) as well.

Reviewer: ```suggestion
It’s important to remember that automated checks can only catch a small number of accessibility issues, therefore we also encourage you to conduct manual testing of your plugins using Assistive technology (screen readers, alternative navigation, and screen magnifiers are a few examples).
```

> If you have identified accessibility issues and don’t have time right now for a contribution, please open an issue over at [Backstage Issues](https://github.com/backstage/backstage/issues) to let us know.

Reviewer: ```suggestion
If you have identified accessibility issues and don’t have time to contribute a fix, please open an issue over at [Backstage Issues](https://github.com/backstage/backstage/issues) to let us know.
```
